### PR TITLE
TCM-2256 Attachments API public docs and SDK

### PR DESCRIPTION
**Pull request by AlexanderNovak**
Replace this text with a description of your proposed change.

[JIRA Ticket](https://bt.smartling.net/browse/{jira_id_here})

## Squash Commit Guidelines

This is a reminder on how to format your sqaush commit message. Delete from your PR after noting below.

### Message Header

The message header is a single line that contains succinct description of the change containing a type: subject. For example:

`feat: add Foo API`

#### Allowed Types

This describes the kind of change that this commit is providing:

- feat (when adding feature)
- fix (when performing a bug fix)
- docs (when adding or changing documentation)
- refactor (when doing non-breaking code refactoring)
- test (when adding missing tests)
- chore (when performing maintainance)

#### Subject Text

This is a very short description of the change.

- use imperative, present tense: “change” not “changed” nor “changes”
- no dot (.) at the end

### CODE REVIEW

**Review by AlexArcher — APPROVED** [review]

**Review by AlexArcher — APPROVED** [review]

File: smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/AttachmentsApi.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+@Consumes(MediaType.APPLICATION_JSON)
+public interface AttachmentsApi
+{
+    @GET
+    @Path("/accounts/{accountUid}/{type}/attachments")
+    ListResponse<AttachmentEntitiesPTO> listAttachments(@PathParam("accountUid") String accountUid, @PathParam("type") String type, @BeanParam AttachmentListFilter listFilter);
+
+    @PUT
+    @Path("/accounts/{accountUid}/{type}")
+    ListResponse<EntityAttachmentsPTO> linkAttachments(@PathParam("accountUid") String accountUid, @PathParam("type") String type, EntityLinkCommand command);
```
Comment (klegins-smartling) [inline smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/AttachmentsApi.java:34]: Is this one admin only rigtht now?  I feel like we can exclude this one from the public docs?

File: smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/AttachmentsApi.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    ListResponse<AttachmentPTO> entityAttachments(@PathParam("accountUid") String accountUid, @PathParam("type") String type, @PathParam("entityUid") String entityUid);
+
+    @POST
+    @Path("/accounts/{accountUid}/{type}")
+    ListResponse<EntityAttachmentsPTO> bulkEntityAttachments(@PathParam("accountUid") String accountUid, @PathParam("type") String type, EntityListCommand command);
+
+    @DELETE
+    @Path("/accounts/{accountUid}/{type}/{entityUid}")
+    EmptyData deleteAttachments(@PathParam("accountUid") String accountUid, @PathParam("type") String type, @PathParam("entityUid") String entityUid);
+
```
Comment (klegins-smartling) [inline smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/AttachmentsApi.java:47]: EmptyData is something we use with deletes elsewhere in in the public api sdk?

File: smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/AttachmentsApi.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    ListResponse<EntityAttachmentsPTO> linkAttachments(@PathParam("accountUid") String accountUid, @PathParam("type") String type, EntityLinkCommand command);
+
+    @GET
+    @Path("/accounts/{accountUid}/{type}/{entityUid}")
+    ListResponse<AttachmentPTO> entityAttachments(@PathParam("accountUid") String accountUid, @PathParam("type") String type, @PathParam("entityUid") String entityUid);
+
+    @POST
+    @Path("/accounts/{accountUid}/{type}")
+    ListResponse<EntityAttachmentsPTO> bulkEntityAttachments(@PathParam("accountUid") String accountUid, @PathParam("type") String type, EntityListCommand command);
+
```
Comment (klegins-smartling) [inline smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/AttachmentsApi.java:43]: I think this one as well can probably be removed from the public apis?

File: smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/AttachmentsApi.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+@Path("/attachments-api/v2")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface AttachmentsApi
+{
+    @GET
+    @Path("/accounts/{accountUid}/{type}/attachments")
+    ListResponse<AttachmentEntitiesPTO> listAttachments(@PathParam("accountUid") String accountUid, @PathParam("type") String type, @BeanParam AttachmentListFilter listFilter);
+
```
Comment (klegins-smartling) [inline smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/AttachmentsApi.java:31]: This one should not be in the public sdk?

File: smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/AttachmentsApi.java (hunk 2 above)
Comment (AlexanderNovak) [reply smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/AttachmentsApi.java:47]: yep,
```
    @POST
    @Path("/projects/{projectId}/file/delete")
    EmptyData deleteFile(@PathParam("projectId") String projectId, DeleteFilePTO deleteFilePTO);
```

File: smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/pto/AttachmentType.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package com.smartling.api.attachments.v2.pto;
+
+public enum AttachmentType
```
Comment (dkBrazz) [inline smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/pto/AttachmentType.java:3]: where is this enum used?
I only found it in test

Comment (AlexanderNovak) [reply smartling-attachments-api/src/main/java/com/smartling/api/attachments/v2/pto/AttachmentType.java:3]: well... this is kind of enum of supported attachment types just for 'visibility' for those who is going to use sdk.

**Review by klegins-smartling — APPROVED** [review]

**Pull request closed without merging**